cdap-ui/app/cdap/components/Wrangler/__tests__/type-inference.test.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {inferType, inferColumn} from 'components/Wrangler/type-inference';
+
+
+describe('Wrangler: Type Inference', () => {
+  it('should infer string', () => {
+    const input1 = 'test 123';
+    const input2 = '123 test';
+
+    expect(inferType(input1)).toBe('string');
+    expect(inferType(input2)).toBe('string');
+  });
+
+  it('should infer booleans', () => {
+    const trueInput = 'true';
+    const falseInput = 'false';
+
+    expect(inferType(trueInput)).toBe('boolean');
+    expect(inferType(falseInput)).toBe('boolean');
+  });
+
+  it('should infer integer', () => {
+    const integerInput = '135';
+    const wrongInteger = '08';
```

Comment:
I checked `inferType` parsing functionality for numbers and thats the reason I am asking how is `08` not a number? `parseFloat(08)` returns `8` and `8%1` is `0` right? So its still a valid number? And wrangler is not strictly for data from a database right? why are we concerned about how it is stored in database?